worker/src/RTC/RtpPacket.cpp

```diff
@@ -566,16 +566,14 @@ namespace RTC
 			return;
 		}
 
-		std::memcpy(extenValue, mid.c_str(), midLen);
-
-		SetExtensionLength(this->midExtensionId, midLen);
+		SetExtensionValue(this->midExtensionId, midLen, mid);
 	}
 
 	/**
-	 * The caller is responsible of not setting a length higher than the
-	 * available one (taking into account existing padding bytes).
+	 *  This method only can set exists extension. if len of value changed, this method will
+	 * recaculate and fill padding.
```

Comment:
```suggestion
	 *  This method set the value for an existing extension. If the length of the current value changed,
	 * it will recalculate it and regenerate the header extension accordingly.
```